server.py

```diff
@@ -51,7 +57,8 @@ def parse_request(request):
         return None
     error_check = check_errors(request_pieces)
     if error_check == 'No Errors':
```

Comment:
Using a string here seems really brittle and prone to failure.  Why not raise a Python error you can use with `try/except`?